### PR TITLE
fix: move setAllConnections to the right statement

### PR DESCRIPTION
**Pull request by likyh**
# Summary

1. move setAllConnections to the right statement (it will cause bug sometimes because `c` is null)
2. replace 2 setters with `useMemo`

### Does this close any open issues?
related #2881

### CODE REVIEW

**Comment by e2corporation** [comment]
This is partially conflicting with #2890, they both have the comment regarding Ln 362 and moving where that var is declared.

**Review by e2corporation — CHANGES_REQUESTED** [review]
Fix conflicts with #2890, resolve issue commented with `c` var, then LGTM

File: config-ui/src/hooks/useConnectionManager.jsx [hunk 1]
```diff
@@ -393,38 +395,32 @@ function useConnectionManager (
           )
           console.log('>> ALL SOURCE CONNECTIONS: ', aC)
         } else {
-          c = await request.get(
+          const c = await request.get(
```
[the remark this answers is not in the document]
Comment (likyh) [reply config-ui/src/hooks/useConnectionManager.jsx:398]: https://github.com/apache/incubator-devlake/pull/2890#discussion_r959407416

[the remark this answers is not in the document]
Comment (e2corporation) [reply config-ui/src/hooks/useConnectionManager.jsx:398]: The `const c` declaration is better defined outside the statement.

**Review by e2corporation — APPROVED** [review]
Nice work, LGTM. `sourceLimits` will have to be refactored again later as it will be injected by the Integrations hook not pulled from the old Providers.js config.

**Pull request merged**